plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ConstantPopulatingPageSource.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.RunLengthEncodedBlock;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.metrics.Metrics;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.OptionalLong;
+import java.util.concurrent.CompletableFuture;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class ConstantPopulatingPageSource
+        implements ConnectorPageSource
+{
+    private final ConnectorPageSource delegate;
+    private final Block[] constantColumns;
```

Comment:
We're blocking here N `Block` instances and `N` `int` delegate indexes for `N` columns.
It is indeed rather straightforward to read, but I think we can do better.

Proposal: a rather naive idea which should work, but adds complexity in figuring out what is happening:
- constantColumns: an array of `B` `Block` corresponding to constant columns
- delegateColumnsIndexes: n array of `D` `int` delegate indexes 
- columnsIndexes: an array of `N` `int` containing the indexes in the block and delegate indexes array
- `B` + `D` = `N`

`columnsIndexes` have the following value structure:
- if pointing to a constant column index, the index value will be between `0` and `constantColumns.size()`
- if pointing to a delegate column index, the index value will be betwen `constantColumns.size()` and `N`

Depending on the value of the index, we know in which array to look for the value we want.